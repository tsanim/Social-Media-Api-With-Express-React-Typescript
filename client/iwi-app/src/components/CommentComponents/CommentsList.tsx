import React from 'react';
import Comment from './Comment';
import CommentForm from '../Forms/CommentForm';
import PropTypes from 'prop-types';
import CommentsListProps from '../../interfaces/Components/CommentComponents/CommentsListProps.interface';

function CommentsList({ comments, currentUser, postId, likeCommentHandler, dislikeCommentHandler, deleteCommentHandler, makeCommentHandler }: CommentsListProps) {
    return (
        <section className="comments">
            <h3>Comments</h3>
            <ul>
                {
                    comments.length > 0
                        ? comments.map(c => (<li key={c._id}>
                            <Comment
                                comment={c}
                                currentUser={currentUser}
                                likeCommentHandler={likeCommentHandler}
                                dislikeCommentHandler={dislikeCommentHandler}
                                deleteCommentHandler={deleteCommentHandler}
                            />
                        </li>))
                        : null
                }
            </ul>
            <CommentForm 
                postId={postId} 
                makeCommentHandler={makeCommentHandler}
            />
        </section>
    )
}

CommentsList.propTypes = {
    comments: PropTypes.array,
    postId: PropTypes.string,
    currentUser: PropTypes.object,
    likeCommentHandler: PropTypes.func,
    makeCommentHandler: PropTypes.func,
    dislikeCommentHandler: PropTypes.func,
    deleteCommentHandler: PropTypes.func
}


export default CommentsList;