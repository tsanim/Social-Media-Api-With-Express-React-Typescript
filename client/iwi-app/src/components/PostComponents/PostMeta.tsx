import React from 'react';
import CommentsList from '../CommentComponents/CommentsList';
import PropTypes from 'prop-types';
import PostMetaProps from '../../interfaces/Components/PostComponents/PostMetaProps.interface';

function PostMeta(props: PostMetaProps) {
    return (
        <div className="postMeta">
            <div className="meta">
                <button onClick={props.handleShowLikesModal} className="likes">{props.likesCount + ' ' + props.likesString}</button>
                <span onClick={props.handleShowComments}>{props.comments.length} Comments</span>
            </div>

            {
                props.areCommentsShown
                    ? <CommentsList
                        comments={props.comments}
                        postId={props.postId}
                        currentUser={props.currentUser}
                        likeCommentHandler={props.likeCommentHandler}
                        dislikeCommentHandler={props.dislikeCommentHandler}
                        deleteCommentHandler={props.deleteCommentHandler}
                        makeCommentHandler={props.makeCommentHandler}
                    />
                    : null
            }

        </div>
    )
}

PostMeta.propTypes = {
    handleShowLikesModal: PropTypes.func,
    handleShowComments: PropTypes.func,
    likesCount: PropTypes.number,
    likesString: PropTypes.string,
    comments: PropTypes.array,
    areCommentsShown: PropTypes.bool,
    postId: PropTypes.string,
    currentUser: PropTypes.object,
    likeCommentHandler: PropTypes.func,
    dislikeCommentHandler: PropTypes.func,
    deleteCommentHandler: PropTypes.func,
    makeCommentHandler: PropTypes.func,
}

export default PostMeta;