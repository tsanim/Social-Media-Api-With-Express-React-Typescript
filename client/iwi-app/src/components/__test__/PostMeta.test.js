import React from 'react';
import PostMeta from '../PostComponents/PostMeta';
import CommentsList from '../CommentComponents/CommentsList';

import { shallow } from 'enzyme';

function postMetaComponentSetup() {
    const props = {
        handleShowLikesModal: jest.fn(),
        handleShowComments: jest.fn(),
        likesCount: 0,
        likesString: '',
        comments: [],
        areCommentsShown: true,
        postId: '',
        currentUser: {},
        likeCommentHandler: jest.fn(),
        dislikeCommentHandler: jest.fn(),
        deleteCommentHandler: jest.fn(),
        makeCommentHandler: jest.fn(),
    }

    const enzymeWrapper = shallow(<PostMeta {...props} />)

    return {
        props,
        enzymeWrapper
    }
}

describe('PostMeta component', () => {
    it('should render initial layout', () => {
        const { enzymeWrapper } = postMetaComponentSetup();

        expect(enzymeWrapper.getElements()).toMatchSnapshot();
    });

    it('should render div tag with class "postMeta"', () => {
        const { enzymeWrapper } = postMetaComponentSetup();

        expect(enzymeWrapper.find('div.postMeta')).toHaveLength(1);
    });

    it('should render div tag with class "meta"', () => {
        const { enzymeWrapper } = postMetaComponentSetup();

        expect(enzymeWrapper.find('div.meta')).toHaveLength(1);
    });

    it('should render a button tag with class "likes"', () => {
        const { enzymeWrapper } = postMetaComponentSetup();

        expect(enzymeWrapper.find('button.likes')).toHaveLength(1);
    });

    it('should render CommentsList Component when areCommentsShown prop is true', () => {
        const { enzymeWrapper } = postMetaComponentSetup();

        expect(enzymeWrapper.find(CommentsList)).toHaveLength(1);
    });
});
