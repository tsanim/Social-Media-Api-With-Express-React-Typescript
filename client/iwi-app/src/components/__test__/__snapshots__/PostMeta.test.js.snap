// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostMeta component should render initial layout 1`] = `
Array [
  <div
    className="postMeta"
  >
    <div
      className="meta"
    >
      <button
        className="likes"
        onClick={[MockFunction]}
      >
        0 
      </button>
      <span
        onClick={[MockFunction]}
      >
        0
         Comments
      </span>
    </div>
    <CommentsList
      comments={Array []}
      currentUser={Object {}}
      deleteCommentHandler={[MockFunction]}
      dislikeCommentHandler={[MockFunction]}
      likeCommentHandler={[MockFunction]}
      makeCommentHandler={[MockFunction]}
      postId=""
    />
  </div>,
]
`;
